gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/CategoryServiceImpl.java

```diff
@@ -107,7 +107,7 @@ public List<CategoryEntity> findByPage(String page) {
     public CategoryEntity findById(final String id, final String environmentId) {
         try {
             LOGGER.debug("Find category by id : {}", id);
-            Optional<Category> category = categoryRepository.findById(id);
+            Optional<Category> category = categoryRepository.findById(id).filter(c -> c.getEnvironmentId().equalsIgnoreCase(environmentId));
             if (!category.isPresent()) {
                 category = categoryRepository.findByKey(id, environmentId);
```

Comment:
should the multi tenancy check also be done here ?

Reply:
no because the findByKey already looks for a category in a specific environment